frame/transaction-payment/src/lib.rs

```diff
@@ -445,6 +445,31 @@ where
 		}
 	}
 
+	/// Query runtime dispatch info of a given `Call`.
+	pub fn query_call_info<Call: GetDispatchInfo>(
```

Comment:
These two associated functions (Pallet::query_call_info, Pallet::query_weight_to_fee), can be moved to `runtime-api.TransactionPayment` since the use case for them is narrow. 
In this case `Pallet::weight_to_fee` visibility needs to be changed to `pub`.
To decide here I need more context on what should be added under Pallet type.

Reply:
Don't create new types. It is customary do just have 1to1 mapping between pallet associated functions and runtime APIs.